### PR TITLE
Comparison

### PR DESCRIPTION
**Pull request by vorosl**

### CODE REVIEW

File: test/wasmBenchmarker/benchmark.py [hunk 1]
```diff
@@ -155,9 +180,9 @@ def compile_tests(emcc_path, path, only_game, only_simd, compile_anyway, run, ve
 
         flags = "-msimd128" if file.startswith("simd") else ""
         flags += (" -s WASM=1 -s EXPORTED_FUNCTIONS=_runtime"
-                  " -s EXPORTED_RUNTIME_METHODS=ccall,cwrap"
-                  " -O2"
-                  f" -o {path}/wasm/{name}.wasm")
+                + " -s EXPORTED_RUNTIME_METHODS=ccall,cwrap"
```
Comment (zherczeg) [inline test/wasmBenchmarker/benchmark.py:183]: Do you need `+` here?

File: .github/workflows/actions.yml [hunk 2]
```diff
@@ -273,7 +273,6 @@ jobs:
     - name: Run Tests
       run: |
         test/wasmBenchmarker/benchmark.py --engines $GITHUB_WORKSPACE/out/linux/x64/walrus --iterations 2 --verbose
-        test/wasmBenchmarker/benchmark.py --engines $GITHUB_WORKSPACE/out/linux/x64/walrus --iterations 2 --jit --verbose
```
Comment (zherczeg) [inline .github/workflows/actions.yml:276]: Is it the default now?

File: test/wasmBenchmarker/benchmark.py [hunk 3]
```diff
@@ -81,7 +81,32 @@ def parse_args():
     parser.add_argument("--jit", help="use JIT version of Walrus", action="store_true")
     parser.add_argument("--verbose", help="prints extra informations e.g. paths", action="store_true")
     parser.add_argument("--no-system-emcc", help="Don't use emcc command from the system", action="store_true", default=False)
-    return parser.parse_args()
+    parser.add_argument("--jit-no-reg-alloc", help="Use JIT without register allocation", action="store_true", default=False)
+    parser.add_argument("--interpreter", help="Use interpreter", action="store_true", default=False)
+    parser.add_argument("--compare-jit-to-interpreter", help="Compare JIT results to interpreter", action="store_true", default=False)
+    parser.add_argument("--compare-jit-no-reg-alloc-to-interpreter", help="Compare JIT no reg alloc results to interpreter", action="store_true", default=False)
+    parser.add_argument("--compare-jit-no-reg-alloc-to-jit", help="Compare JIT no reg alloc results to JIT", action="store_true", default=False)
```
Comment (zherczeg) [inline test/wasmBenchmarker/benchmark.py:88]: Would it be a good idea to provide a list for thisd parameter? This way any combination would be supported (with any number of columns).

File: test/wasmBenchmarker/benchmark.py [hunk 4]
```diff
@@ -281,11 +386,24 @@ def main():
...
+    if not args.no_time:
+        compare(result_data["time"], args.engines, args.compare_jit_to_interpreter, args.compare_jit_no_reg_alloc_to_interpreter, args.compare_jit_no_reg_alloc_to_jit)
+        print("TIME RESULTS")
+        generate_report(result_data["time"], args.report)
     if (args.mem):
+        compare(result_data["mem"], args.engines, args.compare_jit_to_interpreter, args.compare_jit_no_reg_alloc_to_interpreter, args.compare_jit_no_reg_alloc_to_jit)
+        print("MEMORY RESULTS")
         generate_report(result_data["mem"], memreport)
+    
+    print(errorList)
```
Comment (zherczeg) [inline test/wasmBenchmarker/benchmark.py:400]: Perhaps this line could be removed

File: test/wasmBenchmarker/benchmark.py [hunk 5]
```diff
@@ -281,11 +386,24 @@ def main():
...
 
 
+    for d in result_data:
+        for record in result_data[d]:
+            for key in record:
+                if record[key] == -1:
+                    exit(1)
```
Comment (zherczeg) [inline test/wasmBenchmarker/benchmark.py:406]: a global error could be used instead of this

**Review by zherczeg — CHANGES_REQUESTED** [review]

File: .github/workflows/actions.yml (hunk 2 above)
Comment (vorosl) [reply .github/workflows/actions.yml:276]: Default all variation (interpreter, jit, jit-no-reg-alloc) are running, see at: https://github.com/Samsung/walrus/actions/runs/9597089499/job/26465412396

File: test/wasmBenchmarker/benchmark.py (hunk 4 above)
Comment (vorosl) [reply test/wasmBenchmarker/benchmark.py:419]: The empty line or the `print(errorList)`?

Comment (zherczeg) [reply test/wasmBenchmarker/benchmark.py:419]: I think there were 2 empty lines. One is enough

**Comment by vorosl** [comment]
```
TIME RESULTS
|        test        |0 INTERPRETER|0 INTERPRETER/JIT|0 INTERPRETER/JIT_NO_REG_ALLOC|0 JIT|0 JIT/JIT_NO_REG_ALLOC|
|--------------------|-------------|-----------------|------------------------------|-----|----------------------|
|       change       |    4.140    |  1.148 (3.61x)  |         1.366 (3.03x)        |1.148|     1.366 (0.84x)    |
|      factorial     |    6.083    |  0.146 (41.66x) |        0.505 (12.05x)        |0.146|     0.505 (0.29x)    |
|      fannkuch      |    0.598    |  0.050 (11.96x) |         0.120 (4.98x)        |0.050|     0.120 (0.42x)    |
|      fibonacci     |    3.147    |  2.057 (1.53x)  |         2.090 (1.51x)        |2.057|     2.090 (0.98x)    |
|       gregory      |    3.708    |  1.593 (2.33x)  |         1.127 (3.29x)        |1.593|     1.127 (1.41x)    |
|        hanoi       |    6.753    |  0.939 (7.19x)  |         1.834 (3.68x)        |0.939|     1.834 (0.51x)    |
|      heapsort      |    4.614    |  0.433 (10.66x) |         1.289 (3.58x)        |0.433|     1.289 (0.34x)    |
|       huffman      |    4.352    |  0.252 (17.27x) |         1.131 (3.85x)        |0.252|     1.131 (0.22x)    |
|     kNucleotide    |    5.032    |  0.227 (22.17x) |         0.641 (7.85x)        |0.227|     0.641 (0.35x)    |
|  mandelbrotDouble  |    13.752   |  1.257 (10.94x) |         5.717 (2.41x)        |1.257|     5.717 (0.22x)    |
|   mandelbrotFloat  |    13.724   |  1.196 (11.47x) |         3.871 (3.55x)        |1.196|     3.871 (0.31x)    |
|   matrixMultiply   |    4.204    |  0.562 (7.48x)  |         2.229 (1.89x)        |0.562|     2.229 (0.25x)    |
|     miniWalrus     |    2.347    |  0.118 (19.89x) |         0.411 (5.71x)        |0.118|     0.411 (0.29x)    |
|        nbody       |    7.298    |  0.463 (15.76x) |         1.588 (4.60x)        |0.463|     1.588 (0.29x)    |
|       nqueens      |    5.655    |  0.351 (16.11x) |         1.065 (5.31x)        |0.351|     1.065 (0.33x)    |
|        prime       |    3.356    |  0.131 (25.62x) |         0.481 (6.98x)        |0.131|     0.481 (0.27x)    |
|      quickSort     |    5.922    |  0.263 (22.52x) |         1.308 (4.53x)        |0.263|     1.308 (0.20x)    |
|      redBlack      |    5.812    |  0.783 (7.42x)  |         1.920 (3.03x)        |0.783|     1.920 (0.41x)    |
|         rsa        |    3.237    |  0.670 (4.83x)  |         0.931 (3.48x)        |0.670|     0.931 (0.72x)    |
|      salesman      |    6.776    |  0.530 (12.78x) |         1.335 (5.08x)        |0.530|     1.335 (0.40x)    |
|simdMandelbrotDouble|    5.385    |  0.307 (17.54x) |         1.190 (4.53x)        |0.307|     1.190 (0.26x)    |
| simdMandelbrotFloat|    6.092    |  0.195 (31.24x) |         1.526 (3.99x)        |0.195|     1.526 (0.13x)    |
| simdMatrixMultiply |    4.132    |  0.442 (9.35x)  |         1.933 (2.14x)        |0.442|     1.933 (0.23x)    |
|      simdNbody     |    6.952    |  0.240 (28.97x) |         1.044 (6.66x)        |0.240|     1.044 (0.23x)    |
|      ticTacToe     |    4.677    |  0.728 (6.42x)  |         1.091 (4.29x)        |0.728|     1.091 (0.67x)    |


# Means
|  0 INTERPRETER  |0 INTERPRETER/JIT|0 INTERPRETER/JIT_NO_REG_ALLOC|       0 JIT      |0 JIT/JIT_NO_REG_ALLOC|
|-----------------|-----------------|------------------------------|------------------|----------------------|
|5.509919999999999|     14.6688     |             4.48             |0.6032399999999999|  0.42279999999999995 |


# Engines
0: /home/runner/work/walrus/walrus/out/linux/x64/walrus
```

**Comment by zherczeg** [comment]
Nice! I would not create a separate part for means, just add a `-------+-----+...` after the last result, create a new line with 'means' in the `test` column

**Review by zherczeg — APPROVED** [review]
LGTM

**Pull request merged**